autopush/websocket.py

```diff
@@ -163,42 +238,29 @@ def wrapper(result, *w_args, **w_kwargs):
     def base_tags(self):
         """Property that uses None if there's no tags due to a DataDog library
         bug"""
-        return self._base_tags if self._base_tags else None
+        return self.ps._base_tags if self.ps._base_tags else None
 
     def log_err(self, failure, **kwargs):
         """Log a twisted failure out through twisted's log.err"""
         log.err(failure, **kwargs)
 
-    def pauseProducing(self):
```

Comment:
As I understand, they're a part of the IProducer interface, and thus are called by twisted after the Producer is registered. see https://github.com/mozilla-services/autopush/pull/153/files#diff-eff17175be11d1f3b2faddd0c7c93cd9L246 I believe that PushState simply records the state transitions. 

@bbangert, obviously, knows better.